library/public/options.h

```diff
@@ -0,0 +1,107 @@
+/**
+ * @class   options
+ * @brief   Class used to control the different options
+ *
+ * A class to control the different options of f3d.
+ * See the README_libf3d.md for the full listing of options
+ */
+
+#ifndef f3d_options_h
+#define f3d_options_h
+
+#include "export.h"
+
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace f3d
+{
+class F3D_EXPORT options
+{
+public:
+  //@{
+  /**
+   * Default/Copy/move constructors/operators
+   */
+  options();
+  ~options();
+  options(const options& opt);
+  options& operator=(const options& opt) noexcept;
+  options(options&& other) noexcept;
+  options& operator=(options&& other) noexcept;
+  //@}
+
+  class exception : public std::runtime_error
+  {
+  public:
+    exception(const std::string& what = "")
+      : std::runtime_error(what)
+    {
+    }
+  };
+
+  //@{
+  /**
+   * Setters for all supported types
+   */
+  options& set(const std::string& name, bool value);
+  options& set(const std::string& name, int value);
+  options& set(const std::string& name, double value);
+  options& set(const std::string& name, const std::string& value);
+  options& set(const std::string& name, const char* value);
+  options& set(const std::string& name, const std::vector<int>& values);
+  options& set(const std::string& name, const std::vector<double>& values);
+  options& set(const std::string& name, std::initializer_list<int> values);
+  options& set(const std::string& name, std::initializer_list<double> values);
+  //@}
+
+  //@{
+  /**
+   * Reference getters for all supported types
+   */
+  void get(const std::string& name, bool& value) const;
```

Comment:
Is my understanding of these interfaces correct?

```
void get(const std::string& name, bool& value) const;
```
This  "Reference getter" will copy the value into an existing bool. Because this is a copy, any further changes to the passed in parameter after this function will not have any effect.

```
bool& getAsBoolRef(const std::string& name);
```
This "Explicit reference getter" will return a reference to the actual bool underneath, such that doing the following is equivalent to the `toggle` method.

```
auto& b = option.getAsBoolRef();
b = !b;
```

If my understanding is correct, I think the use of the phrase "Reference getter" in both methods is misleading, as it indicates they are doing the same thing, which they are not.

Reply:
You are right. You get it from de function signature but the comment is misleading

Reply:
i will update the doc

Reply:
Adressed in https://github.com/f3d-app/f3d/pull/355